dali/test/python/decoder/test_video.py

```diff
@@ -808,33 +839,118 @@ def decoder_pipeline():
+            filenames=filenames,
+            device=device,
+            sequence_length=sequence_length,
+            stride=stride,
+            enable_frame_num=True,
+        )
+        return videos, frame_no
+
+    pipe = decoder_pipeline(batch_size=batch_size, num_threads=2, device_id=0)
+    (out, frame_no) = pipe.run()
+    assert len(out) > 0, f"No output from decoder pipeline for {codec}"
 
+    for i in range(batch_size):
+        frames = out.as_cpu().at(i)
+        frame_no_cpu = int(frame_no.as_cpu().at(i))
+        assert frames.shape[0] > 0, f"No frames decoded for sample {i}"
+
+        # Get the filename for the i-th sample
+        filename = filenames[i % len(filenames)]
+        reference_frames = extract_frames_from_video(
+            filename, frame_no_cpu, sequence_length, None, stride
+        )
         for frame_idx, frame in enumerate(frames):
-            # Load reference frame using zero-padded frame index
-            ref_frame = reference_frames[frame_idx]
-
-            # Compare frames
-            diff_pixels = np.count_nonzero(np.abs(np.float32(frame) - np.float32(ref_frame)) > 2)
-            total_pixels = frame.size
-            # More than 3% of the pixels differ in more than 2 steps
-            if diff_pixels / total_pixels > 0.03:
-                # Save the mismatched frames for inspection
-                frame_bgr = cv2.cvtColor(frame, cv2.COLOR_RGB2BGR)
-                ref_frame_bgr = cv2.cvtColor(ref_frame, cv2.COLOR_RGB2BGR)
-
-                output_path = f"frame_{frame_idx+1:03d}.png"
-                ref_output_path = f"ref_frame_{frame_idx+1:03d}.png"
-
-                cv2.imwrite(output_path, frame_bgr)
-                cv2.imwrite(ref_output_path, ref_frame_bgr)
-                assert False, (
-                    f"Frame {frame_idx+1} differs from reference by more than 2 steps in "
-                    + f"{diff_pixels/total_pixels*100}% of pixels. "
-                    + f"Expected {ref_frame_bgr} but got {frame_bgr}"
-                )
+            compare_frames(
+                frame, reference_frames[frame_idx], frame_idx, diff_step=diff_step, threshold=0.03
+            )
+
+
+def create_video_all_black_frames(
+    container, codec_fourcc, width=320, height=240, fps=30, duration=1
+):
+    black_frame = np.zeros((height, width, 3), dtype=np.uint8)
+    encoded = None
+    with tempfile.NamedTemporaryFile(suffix=f".{container}", delete=True) as f:
+        fourcc = cv2.VideoWriter_fourcc(*codec_fourcc)
+        out = cv2.VideoWriter(f.name, fourcc, fps, (width, height))
+        for _ in range(int(fps * duration)):
+            out.write(black_frame)
+        out.release()
+        with open(f.name, "rb") as f:
+            encoded = np.frombuffer(f.read(), dtype=np.uint8)
+    assert isinstance(encoded, np.ndarray)
+    return encoded
+
+
+def test_mp4v_video_all_black_frames():
+    skip_if_m60()
+    device = "mixed"
+    container, codec_fourcc = "mp4", "mp4v"
```

Comment:
Isn't `mp4v` a `MPEG-4` video codec we have sample of already in our DALI_extra?

Reply:
You are right. Removed